wremnants/include/muon_calibration.h

```diff
@@ -677,26 +704,126 @@ class JpsiCorrectionsUncHelper {
     std::shared_ptr<const T> correctionHist_;
 };
 
-// jpsi corrections central value for multiple muons
 template <typename T>
-class JpsiCorrectionsRVecHelper : public JpsiCorrectionsHelper<T> {
+class ZNonClosureChargeDepHelper {
 
-using base_t = JpsiCorrectionsHelper<T>;
+public:
+    using hist_t = T;
+    using tensor_t = typename T::storage_type::value_type::tensor_t;
+    using out_tensor_t = Eigen::TensorFixedSize<double, Eigen::Sizes<2>>;
+
+    ZNonClosureChargeDepHelper(T&& corrections) :
+        correctionHist_(std::make_shared<const T>(std::move(corrections))) {}
+
+    // helper for bin lookup which implements the compile-time loop over axes
+    template<typename... Xs, std::size_t... Idxs>
+    const tensor_t &get_tensor_impl(std::index_sequence<Idxs...>, const Xs&... xs) {
+        return correctionHist_->at(correctionHist_->template axis<Idxs>().index(xs)...).data();
+    }
+
+    // variadic templated bin lookup
+    template<typename... Xs>
+    const tensor_t &get_tensor(const Xs&... xs) {
+        return get_tensor_impl(std::index_sequence_for<Xs...>{}, xs...);
+    }
+
+    out_tensor_t operator() (
+        double genQop, float genEta,
+        double recoQop, float recoEta, int recoCharge, double recoPt,  
+        const RVec<float> &cov, double nominal_weight = 1.0,
+        std::vector<std::string> vars =  {"A", "M"}
```

Comment:
Better to do these comparisons with enums than with strings. Or you can  just have three booleans: doA, doM, doE.

Reply:
done [here](https://github.com/WMass/WRemnants/pull/200/commits/5129189730420d818d1254c283a5d5b65aae2ed2)